### PR TITLE
Add custom data type support for Modbus climate

### PR DESCRIPTION
**Pull request by vzahradnik**
## Proposed change

This change adds optional structure parameter into Modbus climate.

## Type of change

- [ ] Spelling, grammar or other readability improvements (`current` branch).
- [ ] Adjusted missing or incorrect information in the current documentation (`current` branch).
- [ ] Added documentation for a new integration I'm adding to Home Assistant (`next` branch).
- [x] Added documentation for a new feature I'm adding to Home Assistant (`next` branch).
- [ ] Removed stale or deprecated documentation.

## Additional information

- Link to parent pull request in the codebase: home-assistant/home-assistant#32439
- This PR fixes or closes issue:

## Checklist

- [x] This PR uses the correct branch, based on one of the following:
  - I made a change to the existing documentation and used the `current` branch.
  - I made a change that is related to an upcoming version of Home Assistant and used the `next` branch.
- [x] The documentation follows the Home Assistant documentation [standards][].

[standards]: https://developers.home-assistant.io/docs/documentation_standards.html

### CODE REVIEW

File: source/_integrations/climate.modbus.markdown [hunk 1]
```diff
@@ -66,6 +66,11 @@ data_type:
   required: false
   type: string
   default: float
+structure:
+  description: "If data_type is custom specify here a double quoted Python struct format string to unpack the value. See Python documentation for details. Ex: >i."
```
Comment (frenck) [inline source/_integrations/climate.modbus.markdown]:
```suggestion
  description: "If `data_type` is custom specified a double-quoted Python struct is expected here, to format the string to unpack the value. See Python documentation for details. Example: `>i`."
```

**Review by frenck — CHANGES_REQUESTED** [review]
Thanks for the PR @vzahradnik!

A small suggestion to make it a bit more readable.

Could you take a look? Thanks! 👍

File: source/_integrations/climate.modbus.markdown (hunk 1 above)
Comment (vzahradnik) [reply source/_integrations/climate.modbus.markdown]: Hi @frenck, I copied a whole block from the docs for `Modbus sensor`. Can I fix the line there as well as part of this PR?

Comment (vzahradnik) [reply source/_integrations/climate.modbus.markdown]: I updated this description also for `Modbus sensor`. It's only a cosmetic change. Please let me know if I should split this into a separate PR. Thanks!

**Review by frenck — DISMISSED** [review]
✅ Approved. Can be merged as soon as the parent PR gets merged.

**Review by frenck — APPROVED** [review]

**Pull request merged**